tuxemon/monster.py

```diff
@@ -544,13 +541,25 @@ def set_level(self, level: int = 5) -> None:
         self.total_experience = self.experience_required()
         self.set_stats()
 
-        # Update moves
+    def set_moves(self, level: int) -> None:
+        """
+        Set monster moves according to the level.
+
+        Parameters:
+            level: The level of the monster.
+
+        """
+        moves = []
         for move in self.moveset:
-            if (
-                move.technique not in (m.slug for m in self.moves)
```

Comment:
Why did you move this check into a new method?

Reply:
for the same reason I explained in the other comment:
- monster,py (it gives the monster all the techniques from 0 to LEVEL);
- set_monster_level.py (it gives the monster all the techniques from LEVEL to NEW_LEVEL)

Reply:
the ex lines 549 and 550 were necessary because there was this (ex lines 317, 318, 319)
```
                if move.level_learned <= self.level:
                    technique = Technique(move.technique)
                    self.learn(technique)
```